Marlin/SanityCheck.h

```diff
@@ -29,6 +29,16 @@
 #define SANITYCHECK_H
 
 /**
+ * Due to the high number of issues related with old versions of Arduino IDE
+ * we are now warning our users to update their toolkits. In a future Marlin
+ * release we will stop supporting old IDE versions and will require user
+ * action to proceed with compilation in such environments.
+ */
+#if defined(ARDUINO) && ARDUINO < 10500
```

Comment:
May I then suggest the following approach ? It will cover both situations, IDE not defining `ARDUINO` and people messing around.

``` cpp
#if !defined(ARDUINO) 
  #error Marlin requires the Arduino libraries to compile.
#elif ARDUINO < 10500
  ...
```